L1Trigger/Phase2L1ParticleFlow/interface/TrackWord.h

```diff
@@ -0,0 +1,135 @@
+#ifndef L1Trigger_Phase2L1ParticleFlow_TrackWord_h
+#define L1Trigger_Phase2L1ParticleFlow_TrackWord_h
```

Comment:
don't we have much of this defined elsewhere for the track trigger? 
Can't one use  DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h
@skinnari @tomalin @mdhildreth

Reply:
Yes, some of the encoding functionality is shared. If I moved to using the TTTrack_TrackWord.h, I would still need to write a derived class because it doesn't contain some of the functionality needed. Right now that's the ability to encode values as fixed point. In later updates that will include using ap_int or ap_fixed types and the ability to return those or an equivalent std::bitset.

Reply:
I don't understand this. Isn't the data content of the track word fixed by the track trigger? 
and if you'd want a version of these accessors that return ap_int or std::bitsets instead of unsigned it, I think it would be better to add it there (or in  a track trigger package if DataFormats are not allowed to depend on HLS)

Reply:
This is going to get quite complex. As far as I can tell, I would to:
1. pick a PF region
2. get the PropagatedTracks from there using the tracks() member function (returns a vector of PropagatedTracks)
3. `auto pftrack = track.src` will return me an l1t::PFTrack pointer (https://github.com/aperloff/cmssw/blob/l1t-phase2-v3.0.2-ImprovePatternFiles/L1Trigger/Phase2L1ParticleFlow/interface/DiscretePFInputs.h#L76)
4. `auto trackRef = pftrack->track()` will return me a TrackRef (https://github.com/cms-sw/cmssw/blob/4bf1289769a572218a3a0732f4a45941e6dcdcdf/DataFormats/L1TParticleFlow/interface/PFTrack.h#L14-L15)
5. `auto val = trackRef.get_xBits()` will return an unsigned int containing the track value we want. However, because this gets me an unsigned int, I would need to convert that to either a std::bitset or an hls ap type.

@skinnari @tomalin @mdhildreth @gpetruc, can you confirm this would be the needed way to get only the bits used by a specific portion of the TTTrack_TrackWord?

Reply:
step 1-4 are just because you're relying on the L1PF to implement the regionalization of the track trigger output.
If the track trigger code provides somewhere an emulation in which the TTrack objects are assigned to the sectors and not  just put in one big vector, then you could directly read that (and it would be much preferrable, since the regionalization currently done by the L1PF code is not likely to match precisely with what the track trigger firmware will do; it's there just as a placeholder of a proper emulation of regional inputs and the CL1 regionizer)